src/com/facebook/buck/go/GoCompileStep.java

```diff
@@ -69,37 +70,45 @@ public GoCompileStep(
 
   @Override
   protected ImmutableList<String> getShellCommandInternal(ExecutionContext context) {
-    ImmutableList.Builder<String> commandBuilder =
-        ImmutableList.<String>builder()
-            .addAll(compilerCommandPrefix)
-            .add("-p", packageName.toString())
-            .add("-pack")
-            .add("-trimpath", workingDirectory.toString())
-            .add("-nolocalimports")
-            .addAll(flags)
-            .add("-o", output.toString());
-
-    for (Path dir : includeDirectories) {
-      commandBuilder.add("-I", dir.toString());
+    ArrayList<String> pathStrings = new ArrayList<>();
+    for (Path path : srcs) {
+      pathStrings.add(path.toString());
     }
+    if (pathStrings.size() > 0) {
+      ImmutableList.Builder<String> commandBuilder =
+          ImmutableList.<String>builder()
+              .addAll(compilerCommandPrefix)
+              .add("-p", packageName.toString())
+              .add("-pack")
+              .add("-trimpath", workingDirectory.toString())
+              .add("-nolocalimports")
+              .addAll(flags)
+              .add("-o", output.toString());
 
-    for (Map.Entry<Path, Path> importMap : importPathMap.entrySet()) {
-      commandBuilder.add("-importmap", importMap.getKey() + "=" + importMap.getValue());
-    }
+      for (Path dir : includeDirectories) {
+        commandBuilder.add("-I", dir.toString());
+      }
 
-    if (asmHeaderPath.isPresent()) {
-      commandBuilder.add("-asmhdr", asmHeaderPath.get().toString());
-    }
+      for (Map.Entry<Path, Path> importMap : importPathMap.entrySet()) {
```

Comment:
Could you rename `importMap` to something that reflects the purpose of this object? It's definitely not a map.